src/hotspot/share/utilities/vmError.cpp

```diff
@@ -725,6 +721,11 @@ void VMError::report(outputStream* st, bool _verbose) {
                    "Runtime Environment to continue.");
     }
 
+  // avoid the cache update for malloc/mmap errors
+  if (should_report_bug(_id)) {
+    os::prepare_native_symbols();
+  }
```

Comment:
Ugh, misuse of "should report bug" as "is oom error" :-/ But no problem, I see you just repeat the existing pattern, that is fine. We can clean that in a separate patch.

Reply:
Hi Thomas, any suggestion for a better name? Maybe there was a reason to keep it generic ?

Reply:
If you want to change this, I would do this:

```
-  static bool should_report_bug(unsigned int id) {
-    return (id != OOM_MALLOC_ERROR) && (id != OOM_MMAP_ERROR);
-  }
+  static bool is_oom_error(unsigned int id) {
+    return (id == OOM_MALLOC_ERROR) || (id == OOM_MMAP_ERROR);
+  }
+  static bool should_report_bug(unsigned int id) {
+    return !is_oom_error(id);
+  }
```